### PR TITLE
Create and use CallingForm enum for standalone samples

### PR DESCRIPTION
**Pull request by vchudnov-g**

### CODE REVIEW

File: src/main/java/com/google/api/codegen/transformer/StaticLangApiMethodTransformer.java [hunk 1]
```diff
@@ -324,7 +334,7 @@ public StaticLangApiMethodView generateRequestObjectAsyncMethod(
         Synchronicity.Async,
         additionalParams,
         methodViewBuilder,
-        Arrays.asList(type));
+        Arrays.asList(CallingForm.RequestAsync));
```
Comment (pongad) [inline src/main/java/com/google/api/codegen/transformer/StaticLangApiMethodTransformer.java]: How do we know this is always `RequestAsync`?

File: src/main/java/com/google/api/codegen/transformer/StaticLangApiMethodTransformer.java [hunk 2]
```diff
@@ -454,7 +464,11 @@ public StaticLangApiMethodView generateAsyncOperationFlattenedMethod(
     methodViewBuilder.exampleName(namer.getAsyncApiMethodExampleName(method));
     methodViewBuilder.callableName(namer.getCallableName(method));
     setFlattenedMethodFields(
-        context, additionalParams, Synchronicity.Async, methodViewBuilder, Arrays.asList(type));
+        context,
+        additionalParams,
+        Synchronicity.Async,
+        methodViewBuilder,
+        Arrays.asList(CallingForm.LongRunningFlattenedAsync));
```
Comment (pongad) [inline src/main/java/com/google/api/codegen/transformer/StaticLangApiMethodTransformer.java]: Similar to above: how do we know this is always flattened and async?

File: src/main/java/com/google/api/codegen/transformer/StaticLangApiMethodTransformer.java [hunk 3]
```diff
@@ -87,7 +88,7 @@ public StaticLangApiMethodView generatePagedFlattenedMethod(
         additionalParams,
         Synchronicity.Sync,
         methodViewBuilder,
-        Arrays.asList(ClientMethodType.PagedFlattenedMethod));
+        Arrays.asList(CallingForm.FlattenedPaging));
```
Comment (pongad) [inline src/main/java/com/google/api/codegen/transformer/StaticLangApiMethodTransformer.java]: FYI, in general `Collections.singletonList` is more efficient, but no need to do it in this PR.

File: src/main/java/com/google/api/codegen/transformer/StaticLangApiMethodTransformer.java (hunk 2 above)
Comment (vchudnov-g) [reply src/main/java/com/google/api/codegen/transformer/StaticLangApiMethodTransformer.java]: We're in method `generateAsyncOperationFlattenedMethod`, so I'm taking the `Async` and `Flattened` at face value, and interpreting `Operation` as `LongRunning`. Are any of these inferred conventions incorrect? If so, we can trace the logic more thoroughly.

File: src/main/java/com/google/api/codegen/transformer/StaticLangApiMethodTransformer.java (hunk 1 above)
Comment (vchudnov-g) [reply src/main/java/com/google/api/codegen/transformer/StaticLangApiMethodTransformer.java]: We're in method `generateRequestObjectAsyncMethod`, so I'm taking the `Request` and `Async` at face value. Is this incorrect? If so, we can trace the logic more thoroughly.

File: src/main/java/com/google/api/codegen/transformer/StaticLangApiMethodTransformer.java (hunk 3 above)
Comment (vchudnov-g) [reply src/main/java/com/google/api/codegen/transformer/StaticLangApiMethodTransformer.java]: Ack.

File: src/main/java/com/google/api/codegen/transformer/StaticLangApiMethodTransformer.java (hunk 1 above)
Comment (pongad) [reply src/main/java/com/google/api/codegen/transformer/StaticLangApiMethodTransformer.java]: I looked through the code a little more. C# seems to also use the `type` for cancellation method. I'm good with this PR, but we might need to revisit.

**Pull request merged**